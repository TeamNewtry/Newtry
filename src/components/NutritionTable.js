import {StyleSheet, View} from 'react-native';
import {Row, Rows, Table, TableWrapper} from 'react-native-table-component';
import React, {useContext} from 'react';
import {LocalizationContext} from './Translations';

const NutritionTable = nut => {
  const {translations} = useContext(LocalizationContext);
  const tableHead = [
    translations['nutrition.table.header1'],
    translations['nutrition.table.header2'],
  ];
  const tableData = [
    [
      translations['nutrition.table.field1'],
      ` ${nut.data?.calories ?? 'undefined'} kcal`,
    ],
    [
      translations['nutrition.table.field2'],
      ` ${nut.data?.totalFat ?? 'undefined'} g`,
    ],
    [
      translations['nutrition.table.field3'],
      ` ${nut.data?.saturatedFat ?? 'undefined'} g`,
    ],
    [
      translations['nutrition.table.field4'],
      ` ${nut.data?.carbohydrates ?? 'undefined'} g`,
    ],
    [
      translations['nutrition.table.field5'],
      ` ${nut.data?.sugar ?? 'undefined'} g`,
    ],
    [
      translations['nutrition.table.field6'],
      ` ${nut.data?.protein ?? 'undefined'} g`,
    ],
    [
      translations['nutrition.table.field7'],
      ` ${nut.data?.salt ?? 'undefined'} g`,
    ],
  ];
  return (
    <View>
      <Table borderStyle={{borderColor: 'transparent'}}>
        <Row
          data={tableHead}
          flexArr={[2, 1]}
          style={styles.head}
          textStyle={styles.textHead}
        />
        <TableWrapper style={styles.wrapper}>
          <Rows
            data={tableData}
            flexArr={[2, 1]}
            resizeMode="contain"
            style={styles.row}
            textStyle={styles.text}
          />
        </TableWrapper>
      </Table>
    </View>
  );
};

export default NutritionTable;

const styles = StyleSheet.create({
  table: {
    borderWidth: 3,
    borderColor: '#EAFFFA',
  },
  head: {
    height: 40,
    backgroundColor: 'rgb(246, 246, 247)',
    marginBottom: 1,
  },
  row: {
    height: 35,
    backgroundColor: 'rgb(246, 246, 247)',
  },
  text: {
    textAlign: 'left',
    fontFamily: 'Comfortaa',
  },
  textHead: {
    textAlign: 'left',
    fontFamily: 'Comfortaa',
    color: 'black',
  },
});
