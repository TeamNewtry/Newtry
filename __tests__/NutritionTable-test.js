import 'react-native';
import React from 'react';
import data from '../__mocks__/productMock';
import NutritionTable from '../src/components/NutritionTable';
import renderer from 'react-test-renderer';

describe('testing Nutrition Table', () => {
  it('renders empty view when no data is passed', () => {
    const tree = renderer.create(<NutritionTable />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders table for passed data', () => {
    const tree = renderer
      .create(<NutritionTable data={data.nutrition} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
