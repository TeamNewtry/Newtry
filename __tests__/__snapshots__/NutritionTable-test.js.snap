// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Nutrition Table renders empty view when no data is passed 1`] = `
<View>
  <View
    style={
      Array [
        undefined,
        Object {
          "borderBottomWidth": 1,
          "borderColor": "gray",
          "borderLeftWidth": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          0,
          Object {
            "flexDirection": "row",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#f1f8ff",
            "height": 40,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "gray",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 2,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
           Average nutrional values
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "gray",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
           per 100g
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "flex": 3,
            },
            0,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Calories
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined kcal
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Fat
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Saturated fat
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Carbohydrates
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Sugar
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Protein
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Salt
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`testing Nutrition Table renders table for passed data 1`] = `
<View>
  <View
    style={
      Array [
        undefined,
        Object {
          "borderBottomWidth": 1,
          "borderColor": "gray",
          "borderLeftWidth": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          0,
          Object {
            "flexDirection": "row",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#f1f8ff",
            "height": 40,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "gray",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 2,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
           Average nutrional values
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "borderColor": "gray",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            false,
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
           per 100g
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "flex": 3,
            },
            0,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Calories
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined kcal
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Fat
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Saturated fat
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Carbohydrates
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Sugar
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Protein
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              0,
              Object {
                "flexDirection": "row",
                "overflow": "hidden",
              },
              Object {
                "height": 35,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 2,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               Salt
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderColor": "gray",
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                },
                Object {
                  "justifyContent": "center",
                },
                undefined,
                undefined,
                Object {
                  "flex": 1,
                },
                false,
                undefined,
              ]
            }
          >
            <Text
              resizeMode="contain"
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ]
              }
            >
               undefined g
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
